src/Http/Http/src/StreamPipeReader.cs

```diff
@@ -35,31 +36,33 @@ public class StreamPipeReader : PipeReader
         private bool _examinedEverything;
         private object _lock = new object();
 
-        private CancellationTokenSource InternalTokenSource
+        /// <summary>
+        /// Creates a new StreamPipeReader.
+        /// </summary>
+        /// <param name="readingStream">The stream to read from.</param>
+        public StreamPipeReader(Stream readingStream)
+            : this(readingStream, minimumSegmentSize: 4096, minimumReadThreshold: 0, MemoryPool<byte>.Shared)
```

Comment:
Why default to 0 instead of say half the minimumSegmentSize?

Reply:
I'm not sure it should be 1/2 by default, actually I'd be worried if we just copied kestrel values by default. Maybe we can just have a fixed minimum buffer size instead of just using 2K. Like 256 bytes or something.

Reply:
> Maybe we can just have a fixed minimum buffer size instead of just using 2K. Like 256 bytes or something.

I'd be okay with that.

Reply:
Though if someone explicitly sets minimumReadThreshold to 0, we should allow that.

Reply:
256 bytes is too small of a default IMO, but it's better than 0.